Views/Actions/components/OffendButton.jsx

```diff
@@ -1,90 +1,163 @@
 import React, { useState } from "react";
-import { StyleSheet, View } from "react-native";
-import { Button, Modal } from "react-native-ui-kitten";
+import {
+  StyleSheet,
+  View,
+  Alert,
+  TouchableOpacity,
+  Text,
+  Modal,
+  TouchableWithoutFeedback
+} from "react-native";
+import normalize from "react-native-normalize";
 
 const offendDictionary = [
   {
     text: "Podważ alternatywność",
-    stats: [-0.5, 0, 0.5, 0.2, 0.3, 0, 0.2, 0.2]
+    stats: [-0.5, 0, 0.5, 0.2, 0.3, 0, 0.2, 0.2],
+    staminaPrice: 50,
+    summary: "Ale jak to alternatywki są już zbyt powszechne?! ;_;"
   },
   {
     text: "Nazwij atencjuszką",
-    stats: [0.5, 0, 0.5, 0.2, 0.2, 0, 0.3, 0.3]
+    stats: [0.5, 0, 0.5, 0.2, 0.2, 0, 0.3, 0.3],
+    staminaPrice: 200,
+    summary:
+      "*wyciąga żyletkę i wstawia na facebooka post o próbie samobójczej*"
   },
   {
     text: "Unfollow na insta",
-    stats: [0, 0, 0.2, 0.2, 0.2, 0, 0.1, 0.1]
+    stats: [0, 0, 0.2, 0.2, 0.2, 0, 0.1, 0.1],
+    staminaPrice: 200,
+    summary: "Ale jak to ten śmieć mnie kurwa nie followuje?!"
   },
   {
     text: "Skrytykuj tatuaż",
-    stats: [0.05, 0, 0.1, 0.1, 0.2, 0, 0.3, 0.1]
+    stats: [0.05, 0, 0.1, 0.1, 0.2, 0, 0.3, 0.1],
+    staminaPrice: 200,
+    summary: "Zajebisty jest, nie zesraj się kurwa! -,-'"
   },
   {
     text: "Obraź Billie",
-    stats: [0.15, 0, 0.2, 0.3, 0.2, 0, 0.1, 0.3]
+    stats: [0.15, 0, 0.2, 0.3, 0.2, 0, 0.1, 0.3],
+    staminaPrice: 200,
+    summary: "Odpierdol się od Billie! -,-,-,-,-'"
   }
 ];
 
 export default function OffendButton(props) {
   const [visible, setVisibility] = useState(false);
 
   const setStats = stats => {
-    props.value.setStats(stats);
+    props.statsValue.setStats(stats);
+  };
+
+  const spendStamina = stamina => {
+    return props.stamina.spendStamina(stamina);
   };
 
   const renderOffendElement = () => (
-    <View>
-      {offendDictionary.map(({ text, stats }) => (
-        <Button
+    <View style={styles.modalContainer}>
+      {offendDictionary.map(({ text, stats, staminaPrice, summary }) => (
+        <TouchableOpacity
           key={text}
-          style={styles.button}
-          status="basic"
+          style={styles.modalButton}
           onPress={() => {
-            setVisibility(!visible);
-            setStats(stats, props.value);
+            setVisibility(false);
+
+            if (spendStamina(staminaPrice)) {
+              setTimeout(() => {
+                setStats(stats, props.statsValue);
+
+                Alert.alert(text, summary, [{ text: "OK ;_;" }], {
```

Comment:
i see you put this button text everywhere... Maybe create a helper function for creating dialogs in your app?